### PR TITLE
Mixed Json Spec and Rust Data #9

### PR DESCRIPTION
**Pull request by ThomAub**
This is to adress issue #9 and not the issue 7 as it was stated in the last commit.
Two parts: 
- a new example in the examples folder
- a refresh of the main readme 
- a new line in the examples readme

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,8 @@
 # Created by https://www.gitignore.io/api/rust,jupyternotebooks,visualstudiocode
 # Edit at https://www.gitignore.io/?templates=rust,jupyternotebooks,visualstudiocode
 
+/
```
Comment (davidB) [inline .gitignore]: why ?

**Review by davidB — CHANGES_REQUESTED** [review]
remove "/" in .gitignore or provide an explication

**Comment by davidB** [comment]
The commit was rejected because the rust code was not formatted with rustfmt

File: .gitignore (hunk 1 above)
Comment (ThomAub) [reply .gitignore]: i don't known why it was there .. it's removed now

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (mockersf) [inline .gitignore:1]: this line was added by gitignore.io, it would be better to not remove it

File: examples/README.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 ## [Cloropleth](https://github.com/procyon-rs/vega_lite_3.rs/blob/master/examples/cloropleth_unemployment.rs)
 
-```
+```rust
```
Comment (mockersf) [inline examples/README.md]: those are actually shell scripts and not rust samples

File: README.md [hunk 4]
```diff
@@ -8,11 +8,17 @@
 [![codecov](https://codecov.io/gh/procyon-rs/vega_lite_3.rs/branch/master/graph/badge.svg)](https://codecov.io/gh/procyon-rs/vega_lite_3.rs)
 [![Dependabot Status](https://api.dependabot.com/badges/status?host=github&repo=procyon-rs/vega_lite_3.rs)](https://dependabot.com)
 
-A Rust api for vega-lite v3. Use it to generate vega-lite json, to display result in your Browser or in Jupyter (via [showata](https://crates.io/crates/showata)).
+## About
```
Comment (mockersf) [inline README.md]: As you're changing the about, could you add a link to Vega itself, and maybe a quick description of what it is?
We're actually never explaining that, and it's bad for a newcomer

File: examples/README.md [hunk 5]
```diff
@@ -95,19 +94,25 @@ Loads data directly from an URL.
...
 cargo run --example from_json_spec
 ```
 
 Load the chart directly from the JSON spec.
 
 <img src="https://raw.githubusercontent.com/procyon-rs/vega_lite_3.rs/master/examples/res/screens/diverging_stacked_bar_chart.png">
 
-# Miscellaneous
+```rust
```
Comment (mockersf) [inline examples/README.md]: this sample is missing a title, a link to the source file and an image

File: examples/README.md [hunk 6]
```diff
@@ -95,19 +94,25 @@ Loads data directly from an URL.
...
+cargo run --example from_mixed_json_rust
+```
+
+Create a chart from existing json and add new data
+
+## Miscellaneous
```
Comment (mockersf) [inline examples/README.md:117]: Miscellaneous title should stay at heading level 1 in markdown

Comment (davidB) [reply examples/README.md:117]: Do not agree, markdown linter check a rule that said that in a markdown file, you could have only one title of level 1, and it should be the first line.

File: README.md (hunk 4 above)
Comment (ThomAub) [reply README.md]: Yes I will do so and also to vega-lite

File: examples/README.md (hunk 6 above)
Comment (ThomAub) [reply examples/README.md:117]: I changed it due to the markdown linter

File: examples/README.md (hunk 3 above)
Comment (ThomAub) [reply examples/README.md]: true

File: .gitignore (hunk 2 above)
Comment (ThomAub) [reply .gitignore:1]: yes

File: examples/README.md (hunk 5 above)
Comment (ThomAub) [reply examples/README.md]: As this was the same type of graph as the ndarray one i didn't feel the need to do it but I agree there should also be a title, link and image.

File: README.md (hunk 4 above)
Comment (davidB) [reply README.md]: Maybe you can remove the title `## About`

**Review by davidB — CHANGES_REQUESTED** [review]
idem Francois, except for the title level of Misc... than should stay to level 2

File: README.md [hunk 7]
```diff
@@ -8,11 +8,11 @@
 [![codecov](https://codecov.io/gh/procyon-rs/vega_lite_3.rs/branch/master/graph/badge.svg)](https://codecov.io/gh/procyon-rs/vega_lite_3.rs)
 [![Dependabot Status](https://api.dependabot.com/badges/status?host=github&repo=procyon-rs/vega_lite_3.rs)](https://dependabot.com)
 
-## About
-
 A Rust API for vega-lite v3 to build chart with a rusty API.
 
-It's possible to generate a complete vega-lite json and display the result in your Browser or in Jupyter. This feature is made possible thanks to [showata](https://crates.io/crates/showata).
+Similar to the [Altair](https://altair-viz.github.io/) project in python, this crate build upon [Vega-Lite](https://vega.github.io/vega-lite/) specifications. Vega-Lite is a high-level grammar of interactive graphics. It provides a concise JSON syntax for rapidly generating visualizations to support analysis. Vega-Lite specifications can be compiled to [Vega](https://vega.github.io/vega/)  specifications. Those specifications are then parsed by Vega’s JavaScript runtime to generate both static images or interactive web-based views.
+This crate has a complete mapping of Vega-LiteV3 specification and can be found in `src/schema.rs`.
```
Comment (davidB) [inline README.md]: Vega-LiteV3 => Vega-Lite 3.4

File: README.md [hunk 8]
```diff
@@ -8,11 +8,11 @@
...
+With all the types and structs, it's possible to create your Rust Vegalite graph that will be serialize into a vega-lite JSON. Thanks to [Showata](https://crates.io/crates/showata) the resulting visualization can be display in your Web-Browser or in a Rust [Jupyter Notebook](https://crates.io/crates/evcxr_jupyter).
```
Comment (davidB) [inline README.md]: try to always use the same syntax for Vega Lite

**Review by davidB — APPROVED** [review]

File: README.md [hunk 9]
```diff
@@ -95,4 +95,4 @@ chart.show()?;
 - [A Dramatic Tour through Python’s Data Visualization Landscape (including ggplot and Altair) – Regress to Impress](https://dsaber.com/2016/10/02/a-dramatic-tour-through-pythons-data-visualization-landscape-including-ggplot-and-altair/)
 - [Specifying Data in Altair — Altair 3.0.0 documentation](https://altair-viz.github.io/user_guide/data.html#long-form-vs-wide-form-data)
 - [Visualization — list of Rust libraries/crates // Lib.rs](https://lib.rs/visualization)
-- [Quicktype](https://quicktype.io/) (got issue with the alternative https://transform.now.sh/json-to-rust-serde) was used to bootstrap `vegalite.rs` from the [vega-lite's json schema](https://vega.github.io/schema/vega-lite/v3.json)
+- [Quicktype](https://quicktype.io/) (got issue with the alternative https://transform.now.sh/json-to-rust-serde) was used to bootstrap `vegalite.rs` from the [vega-lite's json schema](https://vega.github.io/schema/Vega-Lite/v3.json)
```
Comment (davidB) [inline README.md]: search and replace too fast, url are case sensitive. this one no longer works (404)

Comment (ThomAub) [reply README.md]: Yes search and replace too fast .. I’m sorry ! I’ll fix this

**Comment by mockersf** [comment]
> idem Francois, except for the title level of Misc... than should stay to level 2

the three titles level 1 are currently:
- Graph Types, showing how to render various graph types
- Data Sources, showing how to plug various data sources
- Miscellaneous, with currently only one example showing how not to use the builders

Changing Misc to level 2 would put it under Data Sources, which it is not. To do that we would have to add one level to all current titles, and add a title level 1 "Examples" at the top

File: examples/README.md (hunk 6 above)
Comment (mockersf) [reply examples/README.md:117]: to have only one level 1 title, all titles level must be increased then, not just this one

File: README.md [hunk 10]
```diff
@@ -58,4 +95,4 @@ chart.show()?;
 - [A Dramatic Tour through Python’s Data Visualization Landscape (including ggplot and Altair) – Regress to Impress](https://dsaber.com/2016/10/02/a-dramatic-tour-through-pythons-data-visualization-landscape-including-ggplot-and-altair/)
 - [Specifying Data in Altair — Altair 3.0.0 documentation](https://altair-viz.github.io/user_guide/data.html#long-form-vs-wide-form-data)
 - [Visualization — list of Rust libraries/crates // Lib.rs](https://lib.rs/visualization)
-- [Quicktype](https://quicktype.io/) (got issue with the alternative https://transform.now.sh/json-to-rust-serde) was used to bootstrap `vegalite.rs` from the [vega-lite's json schema](https://vega.github.io/schema/vega-lite/v3.json)
+- [Quicktype](https://quicktype.io/) (got issue with the [alternative](https://transform.now.sh/json-to-rust-serde)) was used to bootstrap `vegalite.rs` from the [vega-lite's json schema](https://vega.github.io/schema/vega-lite/v3.json)
```
Comment (mockersf) [inline README.md]: `vegalite.rs` is actually now called `schema.rs`

**Review by mockersf — CHANGES_REQUESTED** [review]

File: README.md (hunk 10 above)
Comment (ThomAub) [reply README.md]: yes nice catch !

**Review by mockersf — APPROVED** [review]

**Comment by mockersf** [comment]
@ThomAub OK for me but could you rebase your branch on top of master ? There are conflicts that can't be rebased automatically

**Comment by ThomAub** [comment]
not sure i did it right ..

**Pull request merged**